craft_application/models/project.py

```diff
@@ -115,6 +124,42 @@ def effective_base(self) -> Any:  # noqa: ANN401 app specific classes can improv
             return self.base
         raise RuntimeError("Could not determine effective base")
 
+    @classmethod
+    def _providers_base(
+        cls, base: str | None  # noqa: ARG003 (unused class method argument)
+    ) -> craft_providers.bases.BaseAlias | None:
+        """Get a BaseAlias from an application's base.
+
+        Applications should override this method to convert an application-specific
+        base to a BaseAlias.
+
+        :returns: The BaseAlias for the base or None if not found.
+        """
+        return None
+
+    @pydantic.root_validator(  # pyright: ignore[reportUnknownMemberType,reportUntypedFunctionDecorator]
+        pre=False
+    )
+    def _validate_devel(cls, values: dict[str, Any]) -> dict[str, Any]:
+        """Validate the build-base is 'devel' for the current devel base."""
+        base = values.get("base")
+        base_alias = cls._providers_base(base)
+
+        # i may need to pass this through `cls._provider_base()` to normalize it because:
+        # charmcraft uses `build-base: ubuntu@devel`
+        # snapcraft and rockcraft use `build-base: devel`
+        build_base = values.get("build_base")
+
+        if base_alias == CURRENT_DEVEL_BASE:
+            if build_base and "devel" in build_base:
```

Comment:
I think we should check `build_base == devel` exactly: my understanding is that this is how all applications define devel build bases

Reply:
Unfortunately, charmcraft uses `build-base: ubuntu@devel` ([source](https://github.com/canonical/charmcraft/blob/b6baa10566e3f3933cbd42392a0fe62cc79d2b6b/tests/unit/models/valid_charms_yaml/full-platforms.yaml#L28)) whereas rockcraft and snapcraft use `build-base: devel`.

I'm considering something like this instead:
```python
build_base_alias = cls._provider_base(build_base)

if build_base_alias == craft_providers.bases.ubuntu.BuilddBaseAlias.DEVEL:
```

Reply:
ah I'm glad you caught this. In this case I agree with the alternative you're suggesting, with the caveat that if `values.get("build_base")` is `None` you should use `base` instead (just for consistency's sake, I don't think it makes a difference in the validator's outcome)

Reply:
Done!